File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.py
class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        freq = [0] * 1001
        for num in arr: freq[num] += 1

        for num in target:
            freq[num] -= 1
            if freq[num] == -1: return False
        
        return True

File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.cs
public class Solution {
    public bool CanBeEqual(int[] target, int[] arr) {
        var freq = new int[1001];
        foreach(var num in arr) freq[num]++;

        foreach(var num in target){
            freq[num]--;
            if(freq[num] == -1) return false;
        }
        return true;
    }
}